### ISSUE TITLE
Missing htcsignet.wd.metrics file and how to inference with WI and WD method

### ISSUE THREAD

**Issue by PhamLongVu** [copycpp/HTCSigNet-Master#1]
Seem like missing the htcsignet.wd.metrics file, and I’ve noticed that the repository is quite complex, particularly when it comes to inference and releasing the code using the WI and WD methods.

![image](https://github.com/user-attachments/assets/d8b145b7-c611-4050-8e25-32f81b3ce36b)

**Comment by PhamLongVu** [comment]
Thank you for your contribution. I really look forward to receiving your response. @copycpp

**Comment by copycpp** [comment]
Thank you very much for your reminder. We have already uploaded the missing file to the repository. Regarding the inference process, we have shared the training weights for download and use. Additionally, please refer to the description in the paper and the README.MD.

**Comment by PhamLongVu** [comment]
Thank you very much for your response. I tried to re-code the inference based on my understanding when using cosine similarity to compare two images (reference and query) with each other. When I infer two genuine signature images from the same person, the distance between them is quite close. However, when I infer the result with one image being a genuine signature and the other being a forgery signature, the distance between these two images is also quite close. I'm not sure if I've done the inference correctly. I would like to send you my inference code. This is the topic I'm really focused on, and I would greatly appreciate your feedback. Once again, thank you for releasing this repo. 

https://drive.google.com/file/d/1Qj9tXqjI_CNJf4688ODcfhnG189LvG-4/view?usp=sharing

**Comment by copycpp** [comment]
I generally understand what you mean. Regarding your usage process, I'd like to point out the following issues: Firstly, if you use the verification system based on WI, then it's necessary to use certain users to fine-tune the model beforehand and then conduct inference. Additionally, if you use the verification system of WD, you can choose to conduct inference directly without fine-tuning. However, the signature images need to undergo data preprocessing both during training and inference. The code is in "https://github.com/copycpp/HTCSigNet-Master/blob/main/HTCSigNet-Master/htcsignet/preprocessing/process_dataset.py".

**Comment by PhamLongVu** [comment]
I noticed that the loss provided in the htcsignet_train.py file is only for the WD classification part. Regarding the WI task, as described in the paper you provided, it relies on distance to compare two features for verification. I believe that comparing two embedding features depends on contrastive losses to work properly, and simply using classification loss for the WI signature verify task is not accurate. Is that correct?
